open-api/rest-catalog-open-api.yaml

```diff
@@ -2800,6 +3057,53 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - options
+      properties:
+        select:
+          description: A list of the selected columns
+          type: array
+          items:
+            type: string
+        project:
+          description: A list of the projections
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        options:
+          type: object
+          additionalProperties:
+            type: string
+        snapshot-id:
+          type: integer
+          format: int64
+        from-snapshot-id:
+          type: integer
+          format: int64
+        to-snapshot-id:
+          type: integer
+          format: int64
+        timestamp-ms:
+          type: integer
+          format: int64
+        case-sensitive:
+          type: boolean
+
+    PlanTableRequest:
+      type: object
+      properties:
+        allOf:
+          $ref: '#/components/schemas/PreplanTableRequest'
```

Comment:
I am not sure if we should directly reference all properties of another request. I'd rather have something like a `PlanContext` that holds all the planning configurations, and let both `PreplanTableRequest` and `PlanTableRequst` have it

Reply:
I can fix this